workflow-to-js-transformer/src/main/resources/Workflow.xslt

```diff
@@ -266,25 +272,61 @@
         document.getField('CAF_ACTION_TO_EXECUTE').clear();
         }
 
-        function recordActionToExecute(document, actionId, actionDetails){
+        function recordActionToExecute(document, actionId){
         document.getField('CAF_ACTION_TO_EXECUTE').add(actionId);
+        }
+    </xsl:template>
+
+    <xsl:template name="utilityFunctions">
+        // evaluate the determined details of an action, either executing the action against document or preparing the
+        // document to execute the action
+        function evaluateActionDetails(document, actionDetails){
+            if(typeof actionDetails === 'function'){
+                actionDetails();
+                recordActionCompletedOrFailed(document);
+                return ALREADY_EXECUTED;
+            }
+            // propagate the post-processing field to response custom data if it exists on the document custom data
+            var responseCustomData = actionDetails.customData ? actionDetails.customData : {};
+            if(!isEmpty(document.getCustomData(postProcessingScriptPropertyName))){
+                responseCustomData[postProcessingScriptPropertyName] = document.getCustomData(postProcessingScriptPropertyName);
+            }
 
-        // propagate the post-processing field to response custom data if it exists on the document custom data
-        var responseCustomData = actionDetails.customData ? actionDetails.customData : {};
-        if(!isEmpty(document.getCustomData(postProcessingScriptPropertyName))){
-        responseCustomData[postProcessingScriptPropertyName] = document.getCustomData(postProcessingScriptPropertyName);
+            // Update document destination queue to that specified by action and pass appropriate settings and customData
+            var queueToSet = !isEmpty(actionDetails.queueName) ? actionDetails.queueName : actionDetails.workerName+"Input";
+            var responseOptions = document.getTask().getResponseOptions();
+            responseOptions.setQueueName(queueToSet);
+            responseOptions.setCustomData(responseCustomData);
+            return ACTION_TO_EXECUTE;
         }
 
-        // Update document destination queue to that specified by action and pass appropriate settings and customData
-        var queueToSet = !isEmpty(actionDetails.queueName) ? actionDetails.queueName : actionDetails.workerName+"Input";
-        var responseOptions = document.getTask().getResponseOptions();
-        responseOptions.setQueueName(queueToSet);
-        responseOptions.setCustomData(responseCustomData);
+        // executes the field mapping action on the document
+        function executeFieldMapping(document, mappings){
+        // get the field values to map (from the document)
+        var documentFieldsValuesMap = {};
+        for(var mappingKey in mappings) {
+        var documentFieldToMapFrom =  document.getField(mappingKey);
+        if(!documentFieldToMapFrom.hasValues()){
```

Comment:
I don't think this `if` block is necessary and in fact I think it might be dangerous, as you're not making the same check when setting the values.

Reply:
Changed to remove if block.